xfuser/model_executor/pipelines/pipeline_hunyuandit.py

```diff
@@ -0,0 +1,795 @@
+import os
+from typing import Callable, Dict, List, Tuple, Callable, Optional, Union
+
+import torch
+import torch.distributed
+from diffusers import HunyuanDiTPipeline
+from diffusers.pipelines.hunyuandit.pipeline_hunyuandit import (
+    SUPPORTED_SHAPE,
+    map_to_standard_shapes,
+    get_resize_crop_region_for_grid,
+    rescale_noise_cfg,
+)
+from diffusers.callbacks import MultiPipelineCallbacks, PipelineCallback
+from diffusers.models.embeddings import get_2d_rotary_pos_embed
+from diffusers.pipelines.stable_diffusion import StableDiffusionPipelineOutput
+
+from xfuser.config import EngineConfig
+from xfuser.logger import init_logger
+from xfuser.distributed import (
+    get_classifier_free_guidance_world_size,
+    get_pipeline_parallel_world_size,
+    get_runtime_state,
+    get_pipeline_parallel_rank,
+    get_cfg_group,
+    get_pp_group,
+    get_sequence_parallel_world_size,
+    get_sp_group,
+    is_dp_last_group,
+    is_pipeline_last_stage,
+    is_pipeline_first_stage,
+)
+from xfuser.model_executor.pipelines import xFuserPipelineBaseWrapper
+from .register import xFuserPipelineWrapperRegister
+
+logger = init_logger(__name__)
+
+@xFuserPipelineWrapperRegister.register(HunyuanDiTPipeline)
+class xFuserHunyuanDiTPipeline(xFuserPipelineBaseWrapper):
+    
+    @classmethod
+    def from_pretrained(
+        cls,
+        pretrained_model_name_or_path: Optional[Union[str, os.PathLike]],
+        engine_config: EngineConfig,
+        **kwargs,
+    ):
+        pipeline = HunyuanDiTPipeline.from_pretrained(
+            pretrained_model_name_or_path, **kwargs
+        )
+        return cls(pipeline, engine_config)
+
```

Comment:
`
    @property
    def guidance_scale(self):
        return self._guidance_scale

    @property
    def guidance_rescale(self):
        return self._guidance_rescale

    # here `guidance_scale` is defined analog to the guidance weight `w` of equation (2)
    # of the Imagen paper: https://arxiv.org/pdf/2205.11487.pdf . `guidance_scale = 1`
    # corresponds to doing no classifier free guidance.
    @property
    def do_classifier_free_guidance(self):
        return self._guidance_scale > 1

    @property
    def num_timesteps(self):
        return self._num_timesteps

    @property
    def interrupt(self):
        return self._interrupt
`

I support it is necessary to copy the @property in HunyuanDiTPipeline to this class